encoding/src/commonMain/kotlin/BitSet.kt

```diff
@@ -4,8 +4,15 @@ public interface BitSet<T> {
     public operator fun get(index: Int): Boolean
     public operator fun set(index: Int, value: Boolean): BitSet<T>
     public fun asPrimitive(): T
+
+    /** Extracts [count] bits at [offset]. e.g. `0b00001100.extract(2, 2) = 0b11 = 0x03`. */
+    public fun extract(offset: Int, count: Int): T
 }
 
+/** Extracts [range] bits . e.g. `0b00001100.extract(2..3) = 0b11 = 0x03`. */
```

Comment:
nit: extra whitespace

```suggestion
/** Extracts [range] bits. e.g. `0b00001100.extract(2..3) = 0b11 = 0x03`. */
```
```suggestion
/** Extracts [range] bits . e.g. `0b00001100.extract(2..3) = 0b11 = 0x03`. */
```